lib/ClangImporter/ClangImporter.cpp

```diff
@@ -358,6 +358,36 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
       // Just use the most feature-rich C language mode.
       "-x", "c", "-std=gnu11",
     });
+
+    // The module map used for Glibc depends on the target we're compiling for,
+    // and is not included in the resource directory with the other implicit
+    // module maps. It's at {freebsd|linux}/{arch}/glibc.modulemap.
+    SmallString<128> GlibcModuleMapPath;
+    if (!importerOpts.OverrideResourceDir.empty()) {
+      GlibcModuleMapPath = importerOpts.OverrideResourceDir;
+    } else if (!searchPathOpts.RuntimeResourcePath.empty()) {
+      GlibcModuleMapPath = searchPathOpts.RuntimeResourcePath;
+    }
+
+    // Running without a resource directory is not a supported configuration.
+    assert(!GlibcModuleMapPath.empty());
+
+    llvm::sys::path::append(
+      GlibcModuleMapPath,
+      swift::getPlatformNameForTriple(triple), triple.getArchName(),
+      "glibc.modulemap");
+
+    // Only specify the module map if that file actually exists.
+    // It may not--for example in the case that
+    // `swiftc -target x86_64-unknown-linux-gnu -emit-ir` is invoked using
+    // a Swift compiler not built for Linux targets.
+    if (llvm::sys::fs::exists(GlibcModuleMapPath)) {
+      invocationArgStrs.insert(invocationArgStrs.end(), {
+        (SmallString<128>("-fmodule-map-file=") + GlibcModuleMapPath).str(),
```

Comment:
I think this is *still* going to go through std::string, but maybe that's all right. I forgot this is building up a vector of std::strings anyway. Let's go back to the Twine version.

Actually, you can simplify it a bit with `push_back`. `insert` is just a trick for adding multiple arguments at once.

````c++
invocationArgStrs.push_back(
    (Twine("-fmodule-map-file=") + GlibcModuleMapPath).str());
````